### PR TITLE
Add support for Typescript files. Closes #12

### LINKED ISSUE #12: Typescript support
I don't know how feasible it is but thought that I should put it out here.
It would be awesome with Typescript support.

### PR DESCRIPTION
**Pull request by gcazaciuc**
Add support for .ts, .tsx parsing and reporting.
A small refactor to allow integration of new parser cleanly
A few tests demonstrating parsing and error reporting in both .ts and .tsx files

### CODE REVIEW

**Comment by ismay** [comment]
Thanks for the pr, looks interesting. I'll try to take a look at it this week 👍

File: package.json [hunk 1]
```diff
@@ -34,7 +34,9 @@
     "eslint-plugin-import": "^2.0.1",
     "jest": "^16.0.2",
     "lodash": "^4.15.0",
-    "stylelint": "^7.3.0"
+    "stylelint": "^7.3.0",
+    "typescript": "^2.3.4",
+    "typescript-eslint-parser": "^3.0.0"
```
Comment (ismay) [inline package.json]: If I'm not mistaken this should be in the dependencies, not devDependencies.

**Comment by ismay** [comment]
Other than `typescript-eslint-parser` not being in the dependencies it looks good to me 👍

**Comment by gcazaciuc** [comment]
I've updated the review. Thanks for pointing that out

**Pull request merged**